index.js

```diff
@@ -0,0 +1,102 @@
+const myLibrary = [
+  {
+    author: 'Jack', title: 'Title of book', pages: 400, hasRead: false,
+  },
+  {
+    author: 'Hoppie', title: 'Ma se poes', pages: 3, hasRead: false,
+  },
+  {
+    author: 'Mary', title: 'A book', pages: 200, hasRead: false,
+  },
+];
+const bookList = document.getElementById('book-list');
+const addBook = document.getElementById('submit');
+
+
+function Book(author, title, pages, hasRead) {
+  this.author = author;
+  this.title = title;
+  this.pages = pages;
+  this.hasRead = hasRead;
+}
+
+function addBookToLibrary(book) {
+  myLibrary.push(book);
+}
+
+function addClickEventForDelete(buttons) {
+  buttons.forEach((btn, index) => {
+    btn.addEventListener('click', () => {
+      myLibrary.splice(index, 1);
+      btn.parentElement.remove();
+    });
+  });
+}
+
+function addClickEventForEdit(buttons) {
+  buttons.forEach((btn, index) => {
+    btn.addEventListener('click', () => {
+      myLibrary[index].hasRead = !myLibrary[index].hasRead;
+      const readElement = document.querySelectorAll('.card')[index].children[3];
+      readElement.innerHTML = `Has read? ${myLibrary[index].hasRead}`;
+    });
+  });
+}
+
+function createBookCard(bookList, book) {
+  const card = bookList.appendChild(document.createElement('div'));
+  card.classList.add('card');
+  card.appendChild(document.createElement('h4')).innerHTML = `${book.title}`;
+  card.appendChild(document.createElement('p')).innerHTML = `Author: ${book.author}`;
+  card.appendChild(document.createElement('p')).innerHTML = `Pages: ${book.pages}`;
+  card.appendChild(document.createElement('p')).innerHTML = `Has read? ${book.hasRead}`;
+  return card;
+}
+
+function createButton(container, btnName) {
+  const btn = container.appendChild(document.createElement('button'));
+  btn.innerHTML = btnName;
+  btn.classList.add(btnName.toLowerCase());
+  btn.classList.add('btn');
+}
+
+for (let i = 0; i < myLibrary.length; i += 1) {
+  const bookCard = createBookCard(bookList, myLibrary[i]);
+  const dltBtn = bookCard.appendChild(document.createElement('button'));
+  dltBtn.innerHTML = 'Delete';
+  dltBtn.classList.add('delete');
+  dltBtn.classList.add('btn');
+  const readBtn = bookCard.appendChild(document.createElement('button'));
+  readBtn.innerHTML = 'Read';
+  readBtn.classList.add('read');
+  readBtn.classList.add('btn');
+}
```

Comment:
- [ ] this should be a function that gets called every time you add/delete/edit a book

Reply:
What you're asking will decrease application performance. If a database were connected, this would also increase the cost unnecessarily for all the CRUD operations. There's a reason the application is programmed the way it is. The request will not be made